charmhub/client.go

```diff
@@ -19,170 +19,61 @@ import (
 	"context"
 	"fmt"
 	"io"
-	"net/http"
 	"net/url"
 	"path"
-	"sort"
 	"strings"
 	"time"
 
 	"github.com/juju/charm/v9"
 	"github.com/juju/errors"
-	"github.com/juju/loggo"
 
 	charmhubpath "github.com/juju/juju/charmhub/path"
 	"github.com/juju/juju/charmhub/transport"
 	charmmetrics "github.com/juju/juju/core/charm/metrics"
-	corelogger "github.com/juju/juju/core/logger"
-	"github.com/juju/juju/version"
 )
 
-// ServerURL holds the default location of the global charm hub.
-// An alternate location can be configured by changing the URL
-// field in the Params struct.
 const (
-	CharmHubServerURL     = "https://api.charmhub.io"
-	CharmHubServerVersion = "v2"
-	CharmHubServerEntity  = "charms"
-
-	MetadataHeader = "X-Juju-Metadata"
+	// DefaultServerURL is the default location of the global Charmhub API.
+	// An alternate location can be configured by changing the URL
+	// field in the Config struct.
+	DefaultServerURL = "https://api.charmhub.io"
 
+	// RefreshTimeout is the timout callers should use for Refresh calls.
 	RefreshTimeout = 10 * time.Second
 )
 
-var (
-	userAgentKey   = "User-Agent"
-	userAgentValue = version.UserAgentVersion
-)
```

Comment:
We still need user agent so charmub can track juju versions etc

Reply:
Yeah, this wasn't removed -- I changed them to `const` and moved them to `http.go` where they're actually used.